ldsc/src/main/resources/install-gc.sh

```diff
@@ -6,35 +6,35 @@
 LDSC_ROOT=/mnt/var/ldsc
 
 # install to the root directory
-mkdir -p "$LDSC_ROOT"
+sudo mkdir -p "$LDSC_ROOT"
 cd "$LDSC_ROOT"
 
 # clone the git repository (
-aws s3 cp s3://dig-analysis-data/bin/ldsc/ldsc-python-3-2023-03-09.zip ./
-unzip ldsc-python-3-2023-03-09.zip -d ./ldsc/
+sudo aws s3 cp s3://dig-analysis-data/bin/ldsc/ldsc-python-3-2023-03-09.zip ./
+sudo unzip ldsc-python-3-2023-03-09.zip -d ./ldsc/
 
 # Download all ancestry specific ldscores into the proper location
 for ANCESTRY in AFR AMR EAS EUR SAS
 do
-  mkdir -p ./ldscore/$ANCESTRY
-  aws s3 cp s3://dig-analysis-data/bin/ldsc/ldscore/ldscore_$ANCESTRY.zip ./ldscore/$ANCESTRY/
-  unzip ./ldscore/$ANCESTRY/ldscore_$ANCESTRY.zip -d ./ldscore/$ANCESTRY/
-  rm ./ldscore/$ANCESTRY/ldscore_$ANCESTRY.zip
+  sudo mkdir -p ./ldscore/$ANCESTRY
+  sudo aws s3 cp s3://dig-analysis-data/bin/ldsc/ldscore/ldscore_$ANCESTRY.zip ./ldscore/$ANCESTRY/
+  sudo unzip ./ldscore/$ANCESTRY/ldscore_$ANCESTRY.zip -d ./ldscore/$ANCESTRY/
+  sudo rm ./ldscore/$ANCESTRY/ldscore_$ANCESTRY.zip
 done
 
 # install dependencies
 sudo yum install -y python3-devel
 
 # reinstall numpy with openblas for multithreading
 sudo yum -y install openblas-devel
-sudo pip3 uninstall -y numpy
-sudo pip3 install numpy
+pip3 uninstall -y numpy
+pip3 install numpy
 
 # install rest of python dependencies
-sudo pip3 install -U bitarray
-sudo pip3 install -U boto3
-sudo pip3 install -U Cython
-sudo pip3 install -U pybind11
-sudo pip3 install -U pythran
-sudo pip3 install -U scipy
-sudo pip3 install -U pandas
```

Comment:
Do you understand why we're adding sudos in places and removing it others?

Reply:
Not really, except that (1) basically everything comes back with a permission denied for regular commands (e.g. aws, or mkdir, or rm, etc.) (2) But if you pip install with a sudo it applies to the wrong version and python will crash claiming to not have the package installed.

Here I copied a file over to GC after it was already run, so those sudo pips wouldn't have worked, I just didn't know it when I committed it at the time I think.

But no, in short, I don't quite know what the permissions are doing here. The jobs are run with the hadoop user, so I think hadoop can't create/delete directories or files, but the if you sudo pip (which you aren't supposed to do) it ultimately isn't installed correctly once you are in EMR 6.7.

Reply:
From EMR 5 -> 6 they changes the user permissions I think so everything got inverted. So you used to not have to sudo anything but pip, but now you sudo everything but pip